metadata/metadata.go

```diff
@@ -151,7 +151,10 @@ func Join(mds ...MD) MD {
 }
 
 type mdIncomingKey struct{}
-type mdOutgoingKey struct{}
+
+// MDOutgoingKey is used as the key for outgoing metadata in ctx if it exists.
+// For gRPC-internal use ONLY.
+type MDOutgoingKey struct{}
```

Comment:
Please move this to `internal/metadata` and reference it from there.  Ideally users should never see any of our internal symbols at all.